src/QsCompiler/CompilerOptimizations/Utils.fs

```diff
@@ -237,3 +226,15 @@ let internal longPow (a: int64) (b: int64): int64 =
         power <- power >>> 1
     returnValue
 
+
+/// Returns the intial part of the list that satisfies the given condition, just as List.takeWhile().
+/// Also returns the first item that doesn't satisfy the given condition, or if all items satisfy the condition, returns None.
+let rec internal takeWhilePlus1 (f: 'A -> bool) (l : list<'A>) =
```

Comment:
takeWhileTrue woud be a better name

Reply:
This function is like the existing function takeWhile, but also returns 1 additional element of the list, hence takeWhilePlus1. I'm not sure takeWhileTrue is a better name, as it sounds very similar to takeWhile. I agree the current name isn't great, do you have any other suggestions for what to rename it to?

Reply:
anything else get's pretty long, so let's leave it until someone has a better idea.